Modix.Data.Test/Repositories/EmojiRepositoryTests.cs

```diff
@@ -0,0 +1,300 @@
+        {
+            const int RecordCount = 100;
+
+            (var modixContext, var uut) = BuildTestContext();
+
+            await uut.CreateMultipleAsync(data, RecordCount);
+
+            var emojis = modixContext.Emoji
+                .OrderByDescending(x => x.Id)
+                .Take(RecordCount)
+                .ToArray();
+
+            foreach (var emoji in emojis)
+            {
+                emoji.GuildId.ShouldBe(data.GuildId);
+                emoji.ChannelId.ShouldBe(data.ChannelId);
+                emoji.MessageId.ShouldBe(data.MessageId);
+                emoji.UserId.ShouldBe(data.UserId);
+                emoji.EmojiId.ShouldBe(data.EmojiId);
+                emoji.EmojiName.ShouldBe(data.EmojiName);
+                emoji.IsAnimated.ShouldBe(data.IsAnimated);
+                emoji.Timestamp.ShouldBeInRange(
+                    DateTimeOffset.Now - TimeSpan.FromSeconds(1),
+                    DateTimeOffset.Now + TimeSpan.FromSeconds(1));
+                emoji.UsageType.ShouldBe(data.UsageType);
+            }
+
+            modixContext.Emoji.Where(x => !emojis.Select(y => y.Id).Contains(x.Id)).Select(x => x.Id).ShouldBe(Emoji.Entities.Select(x => x.Id));
+            modixContext.Emoji.Where(x => !emojis.Select(y => y.Id).Contains(x.Id)).EachShould(x => x.ShouldNotHaveChanged());
+
+            await modixContext.ShouldHaveReceived(1)
+                .SaveChangesAsync();
+        }
+
+        #endregion CreateMultipleAsync() Tests
+
+        #region DeleteAsync() Tests
+
+        [TestCaseSource(nameof(ExceptionDeleteTestCases))]
+        public async Task DeleteAsync_CriteriaIsInvalid_Throws(EmojiSearchCriteria criteria, Type exceptionType)
+        {
+            (var modixContext, var uut) = BuildTestContext();
+
+            await Should.ThrowAsync(() => uut.DeleteAsync(criteria), exceptionType);
+        }
+
+        [TestCaseSource(nameof(NonexistentDeleteTestCases))]
+        public async Task DeleteAsync_Nonexistent_DoesNotDeleteEmojis(EmojiSearchCriteria criteria)
+        {
+            (var modixContext, var uut) = BuildTestContext();
+
+            await uut.DeleteAsync(criteria);
+
+            modixContext.Emoji.Count().ShouldBe(Emoji.Entities.Count());
+            modixContext.Emoji.Select(x => x.Id).ShouldBe(Emoji.Entities.Select(x => x.Id));
+            modixContext.Emoji.EachShould(x => x.ShouldNotHaveChanged());
+        }
+
+        [TestCaseSource(nameof(ValidDeleteTestCases))]
+        public async Task DeleteAsync_ValidCriteria_DeletesEmojis(EmojiSearchCriteria criteria, int deletedCount)
+        {
+            (var modixContext, var uut) = BuildTestContext();
+
+            await uut.DeleteAsync(criteria);
+
+            modixContext.Emoji.Count().ShouldBe(Emoji.Entities.Count() - deletedCount);
+            modixContext.Emoji.EachShould(x => x.ShouldNotHaveChanged());
+        }
+
+        #endregion DeleteAsync() Tests
+
+        #region Single GetEmojiStatsAsync() Tests
```

Comment:
Nit/FYI

Usually, when working with overloads, I qualify them by listing what the param types or names are (at least, the important ones). 

For this, I would probably do `GetEmojiStatsAsync(emoji)` with `GetEmojiStatsAsync_Emoji_` and `GetEmojiStatsAsync(emojiIds)` with `GetEmojiStatsAsync_EmojiIds`.